web/Controllers/PropertyController.cs

```diff
@@ -151,7 +139,55 @@ public ActionResult ViewProperty(int propertyId)
             _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
             return RedirectToAction("Error", "Home", new { status = 404 });
         }
-        var propertyViewModel = PropertyViewModel.FromDbModel(model);
+        PropertyViewModel propertyViewModel = PropertyViewModel.FromDbModel(model);
         return View(propertyViewModel);
     }
+
+    [HttpGet]
+    public async Task<IActionResult> ListPropertyImages(int propertyId)
+    {
+        List<string> fileNames = await _azureStorage.ListFilesAsync("property", propertyId);
+        return View(fileNames);
+    }
+    
+    [Authorize(Roles = "Landlord, Admin")]
+    [HttpPost("addImage")]
+    public async Task<IActionResult> AddPropertyImages(List<IFormFile> images, int propertyId)
+    {
+        var user = GetCurrentUser();
+        if (!user.IsAdmin && _propertyService.GetPropertyByPropertyId(propertyId)!.LandlordId != user.LandlordId)
+        {
+            return StatusCode(403);
+        }
+        foreach (var image in images)
+        {
+            if (image.Length > 0)

```

Comment:
When does this happen (length is zero)?

Reply:
I believe the length is zero if the file uploaded (through some sort of glitch) contains no data - it seemed sensible to not allow the upload of an empty/damaged file.

Reply:
Fair enough. Maybe log a warning in this case just so we can see when it happens.